src/components/profile-page/access-tokens/access-token-creation-form.tsx

```diff
@@ -0,0 +1,149 @@
+    return {
+      min: today.toISODate(),
+      max: today
+        .plus({
+          year: 2
+        })
+        .toISODate(),
+      default: today
+        .plus({
+          year: 1
+        })
+        .toISODate()
+    }
+  }, [])
+
+  const tokenDataInitialState = useMemo(() => {
+    return {
+      expiryDate: expiryDates.default,
+      label: ''
+    }
+  }, [expiryDates])
+
+  const [newTokenData, setNewTokenData] = useState<NewTokenFormValues>(() => tokenDataInitialState)
+  const [newTokenWithSecret, setNewTokenWithSecret] = useState<AccessTokenWithSecret>()
+
+  const onHideCreatedModal = useCallback(() => {
+    setNewTokenData(tokenDataInitialState)
+    setNewTokenWithSecret(undefined)
+  }, [tokenDataInitialState])
+
+  const onChangeTokenLabel = useCallback((event: ChangeEvent<HTMLInputElement>) => {
+    setNewTokenData((previousData) => {
+      return {
+        ...previousData,
+        label: event.target.value
+      }
+    })
+  }, [])
+
+  const onChangeTokenExpiry = useCallback((event: ChangeEvent<HTMLInputElement>) => {
+    setNewTokenData((previousData) => {
+      return {
+        ...previousData,
+        expiryDate: event.target.value
+      }
+    })
+  }, [])
+
+  const onCreateToken = useCallback(
+    (event: FormEvent<HTMLFormElement>) => {
+      event.preventDefault()
+      postNewAccessToken(newTokenData.label, newTokenData.expiryDate)
+        .then((tokenWithSecret) => {
+          setNewTokenWithSecret(tokenWithSecret)
+        })
+        .catch(showErrorNotification('profile.accessTokens.creationFailed'))
+    },
+    [newTokenData]
+  )
+
+  const newTokenCreatable = useMemo(() => {
+    return newTokenData.label.trim() !== ''
+  }, [newTokenData])
+
+  return (
+    <Fragment>
+      <h5>
+        <Trans i18nKey={'profile.accessTokens.createToken'} />
+      </h5>
+      <Form onSubmit={onCreateToken} className='text-start'>
+        <Form.Group>
+          <Form.Label>
+            <Trans i18nKey={'profile.accessTokens.label'} />
+          </Form.Label>
+          <Form.Control
+            type='text'
+            size='sm'
+            placeholder={t('profile.accessTokens.label')}
+            value={newTokenData.label}
+            className='bg-dark text-light'
+            onChange={onChangeTokenLabel}
+            isValid={newTokenCreatable}
+            required
+            {...cypressId('access-token-add-input-label')}
+          />
+        </Form.Group>
+        <Form.Group>
+          <Form.Label>
+            <Trans i18nKey={'profile.accessTokens.expiry'} />
+          </Form.Label>
+          <Form.Control
+            type='date'
+            size='sm'
+            value={newTokenData.expiryDate}
+            className='bg-dark text-light'
+            onChange={onChangeTokenExpiry}
+            min={expiryDates.min}
+            max={expiryDates.max}
+            required
+            {...cypressId('access-token-add-input-expiry')}
+          />
+        </Form.Group>
```

Comment:
:banana: 

The form controls (and the following button) are good candidates for component extractions. It would improve the readablity of the whole component because it reduces the loc in this file. You could also extract stuff like onCreateToken into custom hooks.